Routes and http

SharpP03 opened the pull request.

Commit:
Added NavBar component for navigation.
+imported logo
+example page "dashboard"

Ereffe commented on src/components/NavBar.tsx:
```diff
@@ -0,0 +1,30 @@
+import "./NavBar.css";
+import logo from "../assets/logo.png";
+
+function NavBar() {
```
Tutaj jest idealne miejsce żeby dodać {children} :)

Dodaj NavBar({children}: {children:{ React.ReactNode}})

Ereffe commented on src/components/NavBar.tsx:
```diff
@@ -0,0 +1,30 @@
+          <a href="/register">Register</a>
+          <a href="/dashboard">Dashboard</a>
+        </div>
+      </header>
```
Za header wrzuć {children}

Ereffe commented on src/pages/dashboard.tsx:
```diff
@@ -0,0 +1,26 @@
+function dashboard() {
```
Import navbar

Ereffe commented on src/pages/dashboard.tsx:
```diff
@@ -0,0 +1,26 @@
+function dashboard() {
+  return (
+    <div className="dashboard_container">
```
I otocz tego całego diva navbarem

Commit:
updated NavBar to display on dashboard only.

Commit:
New page for Chat
+chat.tsx form that accepts roomKey:string and username:string
+-rename .css for pages
+added route to caht in NavBar

Commit:
fix for NavBarProps type

Ereffe commented on src/App.css:
```diff
@@ -5,6 +5,8 @@
   text-align: center;
 }
 
+.Nav
```
czm to jest puste

SharpP03 replied:
Do usunięcia.

Ereffe commented on src/App.tsx:
```diff
@@ -1,18 +1,30 @@
+import Dashboard from "./pages/dashboard";
+import Chat from "./pages/chat";
+// import Layout from "./components/Layout";
+// import NavBar from "./components/NavBar"; --> imported in dashboard
```
usuń nieużywane importy

SharpP03 replied:
zrobione.

Ereffe commented on src/App.tsx:
```diff
@@ -1,18 +1,30 @@
+// import NavBar from "./components/NavBar"; --> imported in dashboard
 
+import { BrowserRouter, Routes, Route } from "react-router";
+// import { Children } from "react";
```
usuń nieużywane importy

Ereffe commented on src/components/NavBar.tsx:
```diff
@@ -0,0 +1,39 @@
+  children?: React.ReactNode;
+};
+
+function NavBar({ children }: NavBarProps) {
```
usuń children

SharpP03 replied:
usunięte children i zbędne importy

Ereffe commented on src/components/NavBar.tsx:
```diff
@@ -0,0 +1,39 @@
+          <a href="/chat">Chat Room</a>
+        </div>
+      </header>
+      {children}
```
usuń children

Commit:
Fixes.

Ereffe committed:
Merge branch 'main' into routesAndHttp
